File: CS50X/Python/sentimental-readability/readability.py
from cs50 import get_string

puntuacion = '!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~'
frase = get_string("Frase : ")
letras = 0.00
frases = 0.00
palabras = 1.0

for n, char in enumerate(frase):
    if (frase[n] != ' ' and frase[n].isalpha()):
        letras += 1
    elif (frase[n] == '.' or frase[n] == '?' or frase[n] == '!') and (n == len(frase)-1 or frase[n+1].isspace() or frase[n+1] in puntuacion):
        frases += 1
    elif (frase[n].isspace() or n == len(frase) - 1):
        palabras += 1

L = (letras*100) / palabras
S = (frases*100) / palabras

index = round(0.0588 * L - 0.296 * S - 15.8)

if index > 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

File: CS50X/C/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    float letras = 0.0;
    float frases = 0.0;
    float palabras = 1.0;
    float L, S;
    int index;
    string texto = get_string("Text: ");

    for (int x = 0; x < strlen(texto); x++)
    {
        if (texto[x] != ' ' && isalpha(texto[x]))
        {
            letras++;
        }
        else if ((texto[x] == '.' || texto[x] == '?' || texto[x] == '!') &&
                 (isspace(texto[x + 1]) || ispunct(texto[x + 1]) || x == strlen(texto) - 1))
        {
            frases++;
        }
        else if (isspace(texto[x]) || x == strlen(texto) - 1)
        {
            palabras++;
        }
    }

    L = (letras * 100) / palabras;
    S = (frases * 100) / palabras;

    index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}
